File: file_copy_increment.js
const fs = require('fs'); // Import the file system module
const path = require('path'); // Import the path module

const dirPath = '/Users/johndoe/Git/LexDAO/LexDAO_Membership_Token/Image_Files'; // Set the directory path
const nTimes = 5; // Set the number of times to copy each file

function doCopy(dirPath, nTimes) {
  if (!fs.existsSync(dirPath)) { // Check if the directory path exists
    throw new Error('Directory path does not exist'); // Throw an error if the directory path does not exist
  }

  process.chdir(path.join(process.cwd(), dirPath)); // Change the current working directory to the specified directory path

  for (const file of fs.readdirSync(process.cwd())) { // Loop through each file in the directory
    if (!fs.statSync(file).isFile() || file[0] === '.') { // Check if the file is a regular file and not a hidden file
      continue; // Skip the file if it's not a regular file or it's a hidden file
    }

    const { name, ext } = path.parse(file); // Get the filename and extension of the file

    console.log('This is the original file name:', name); // Log the original filename to the console

    const startInd = parseInt(name, 10); // Parse the filename as an integer

    for (let n = 1; n <= nTimes; n++) { // Loop nTimes times
      const newFile = path.join(process.cwd(), `${startInd + n}${ext}`); // Set the new filename
      fs.copyFileSync(path.join(process.cwd(), file), newFile); // Copy the file with the new filename
    }
  }
}

doCopy(dirPath, nTimes); // Call the doCopy function with the specified directory path and number of times to copy each file

File: file_copy_increment.py
#note that this is a Python script that requires a seed file called '0.[extension]'.  For this example
#the file uses a 0.json file

import os
import shutil

dir_path ='/Users/johndoe/Git/LexDAO/LexDAO_Membership_Token/Image_Files' 
n_times = 250



def do_copy(dir_path, n_times):
    assert os.path.exists(dir_path)
    os.chdir(os.path.join(os.getcwd(), dir_path))
    for file in os.listdir(os.curdir):
        if not os.path.isfile(file) or file [0] == ".":  # only change names if it's a file or not a .file
            continue
        name, ext = os.path.splitext(file)
        print("This is the original file name: ", name)
        start_ind = int(name)
        for n in range(1, n_times+1):
            new_file = os.path.join(os.getcwd(), f"{start_ind+n}{ext}") #may want to remove the extension bc of how things get minted
            shutil.copy(os.path.join(os.getcwd(), file), new_file)

do_copy(dir_path, n_times) 



